js/private/js_run_devserver.bzl

```diff
@@ -51,30 +70,20 @@ def _js_run_devserver_impl(ctx):
 
     default_data_runfiles = [target[DefaultInfo].default_runfiles.files for target in ctx.attr.data]
 
-    # The .to_list() calls here are intentional and cannot be avoided; they should be small sets of
-    # files as they only include direct npm links (node_modules/foo) and the package store tree
-    # artifacts those symlinks point to (node_modules/.aspect_rules_js/foo@1.2.3/node_modules/foo)
-    data_files = []
-    for f in depset(transitive = transitive_runfiles + [dep.files for dep in ctx.attr.data] + default_data_runfiles).to_list():
-        if "/.aspect_rules_js/" in f.path:
-            # Special handling for package store deps; we only include 1st party deps since copying
-            # all 3rd party node_modules over is expensive for typical graphs
-            path_segments = f.path.split("/")
-            package_name_segment = path_segments.index(".aspect_rules_js") + 1
-
-            # TODO: @0.0.0 is by default the version of all 1p linked packages, however, it can be overridden by users
-            # if they are manually linking a 1p package and not using workspace. A more robust solution would be to
-            # split handling of 1p and 3p package in the JsInfo provider itself. Other optimizations in the rule set
-            # could also be made if that was the case.
-            if len(path_segments) > package_name_segment and "@0.0.0" in path_segments[package_name_segment]:
-                # include this first party linked dependency
-                data_files.append(f)
-        else:
-            data_files.append(f)
+    # Build the list of data files to copy to the custom sandbox using ctx.actions.args()
+    entries = ctx.actions.args()
+    entries.set_param_file_format("multiline")
+    entries.add("[")
```

Comment:
Why not just do a newline-separats list? Do we really worry about newlines in filenames?

Reply:
Good question. Mainly because json is easy though, and I'm copying `js_image_layer`.

However I'd like to explore streaming this file content instead of doing a full read + `JSON.parse` so we can start the ugly file-syncing without having to wait for this potentially-giant json file to reparse...

Reply:
I might try this later, for now this change requires hardly any change to the .mjs file so I'd like to stick with this for now...